app/helpers/link_helper.rb

```diff
@@ -122,6 +122,11 @@ def generate_icon_prefix(icon_name, prefix)
     return "<span class=\"no-wrap\"><i class=\"fas fa-#{icon_name}\" aria-hidden=”true”></i> #{prefix}</span>".html_safe
   end
 
+  def view_page_on_main_site_button
+    new_url = request.original_fullpath.sub('admin/', '')
+    link_to("View on Main Site", new_url, class: 'btn btn-secondary') if new_url != request.original_fullpath
```

Comment:
i thought of a fun edge case here, if the slug ends in admin, this messes up the url and the check

Reply:
Fixed now!